authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -198,34 +194,212 @@ func getActionName(verb string) string {
+			authInfoSingle.AuthorizationEntity.Id = path.Join(authInfoSingle.AuthorizationEntity.Id, action)
+			authInfoList = append(authInfoList, authInfoSingle)
+
+		} else if subRevReq.ResourceAttributes.Resource == "*" {
+			/*
+				    This sections handles the following scenario's:
+
+				    | Scenario              | Namespace is empty (Cluster scope call)                    | Namespace is not empty (NS scope)             |
+					------------------------ ------------------------------------------------------------  ----------------------------------------------
+					| Verb-*, Res-*, Group-*| All cluster and ns res with all verbs at clusterscope      | All ns resources at ns scope                  |
+
+					| Res-*, Group-*        | All cluster and ns res with specified verb at clusterscope | All ns res with specified verb at ns scope    |
+
+					| Verb-*, Res-*         | All cluster and ns res with all verbs under                | All ns res with all verbs under specified     |
+					|                       | specified apigroup at clusterscope                         | apigroup at nsscope                           |
+
+					| Resource - *          | All CS and NS Resources under specifed apigroup with       | All NS Resources under specifed apigroup with |
+					|                       | specified verb at cluster scope                            | specified verb at ns scope                    |
+			*/
+
+			filteredOperations := make(map[string][]map[string]map[string]oputil.DataAction)
+			if subRevReq.ResourceAttributes.Group == "*" {
+				// all resources under all apigroups
+				filteredOperations = dataActionsMap
+			} else if subRevReq.ResourceAttributes.Group != "" {
+				// all resources under specified apigroup
+				if value, found := dataActionsMap[subRevReq.ResourceAttributes.Group]; found {
+					filteredOperations[subRevReq.ResourceAttributes.Group] = value
+				} else {
+					return nil, errors.New(fmt.Sprintf("No resources found for group %s", subRevReq.ResourceAttributes.Group))
+				}
+			} else {
+				// if Group is not there that means it is the core apigroup
+				if value, found := dataActionsMap["v1"]; found {
+					filteredOperations["v1"] = value
+				} else {
+					return nil, errors.New("No resources found for the core group")
+				}
+			}
+
+			// if Namespace is populated, we need to create the list only for namespace scoped resources
+			finalFilteredOperations := make(map[string][]map[string]map[string]oputil.DataAction)
+
+			if subRevReq.ResourceAttributes.Namespace != "" {
+				for group, resMapList := range filteredOperations {
+					for _, resValues := range resMapList {
+						for _, verbValues := range resValues {
+							for _, dataAction := range verbValues {
+								if dataAction.IsNamespacedResource {
+									finalFilteredOperations[group] = append(finalFilteredOperations[group], resValues)
+									break
```

Comment:
i read it multiple times and am not sure if this `break` is breaking the right loop?